R/radab.R

```diff
@@ -1,20 +1,28 @@
-#' Anti-drug Antibody Analysis Dataset (ADAB)
+#' Anti-Drug Antibody Analysis Dataset (ADAB)
 #'
-#' @param constants (`character vector`)\cr Constant parameters to be used for creating analysis values.
-#' @inheritParams radpc
-#' @param ADPC (`character`)\cr parameter code.
-#' @param paramcd (`character`)\cr parameter code.
-#' @param param (`character`)\cr parameters.
+#' @description `r lifecycle::badge("stable")`
+#'
+#' Function for generating a random Anti-Drug Antibody Analysis Dataset for a given
+#' Subject-Level Analysis Dataset and Pharmacokinetics Analysis Dataset.
 #'
-#' @details One record per study per subject per parameter per time point
+#' @inheritParams argument_convention
+#' @inheritParams radpc
+#' @param ADPC (`data.frame`)\cr Pharmacokinetics Analysis Dataset.
+#' @template param_cached
+#' @templateVar data adab
```

Comment:
```suggestion
```
These are no more needed  right?

Reply:
Why did you keep param_cached? also, I could not find any explanation on how templateVar actually works... does it bind it to the internal dataset?

Reply:
@Melkiades param_cached is a template for just the `@param cached` documentation which allows for customized text within the template. The templateVar is used by the template to determine what value should be used to customized the text in the `@param cached` part. I kept it in since the output was relatively clean.

For example, in `radsl` templateVar sets `data = "adsl"`, and then the `data` argument is implemented in param_cached to output the customized description: 
"whether the cached **ADSL** data **cadsl** should be returned or new data should be generated. If set to TRUE then the other arguments to **radsl** will be ignored."

Reply:
fantastic! now I understood. Thanks, Emily for the explanation. It is actually cool like this